crates/modelardb_compression/src/models/mod.rs

```diff
@@ -49,28 +49,25 @@ pub(super) const VALUE_SIZE_IN_BYTES: u8 = mem::size_of::<Value>() as u8;
 /// Size of [`Value`] in bits.
 pub(super) const VALUE_SIZE_IN_BITS: u8 = 8 * VALUE_SIZE_IN_BYTES;
 
-/// General error bound that is guaranteed to not be negative, infinite, or NAN.
-/// For [`PMCMean`] and [`Swing`] the error bound is interpreted as a relative
-/// per value error bound in percentage, while [`Gorilla`] uses lossless
-/// compression.
+/// Error bound in percentage that is guaranteed to be from 0.0% to 100.0%. For both [`PMCMean`],
+/// [`Swing`], and [`Gorilla`] the error bound is interpreted as a relative per value error bound.
 #[derive(Debug, Copy, Clone)]
 pub struct ErrorBound(f32);
 
 impl ErrorBound {
-    /// Return [`ErrorBound`] if `error_bound` is a positive finite value,
-    /// otherwise [`CompressionError`](ModelarDbError::CompressionError).
-    pub fn try_new(error_bound: f32) -> Result<Self, ModelarDbError> {
-        if error_bound < 0.0 || error_bound.is_infinite() || error_bound.is_nan() {
+    /// Return [`ErrorBound`] if `percentage` is a value from 0.0% to 100.0%, otherwise
+    /// [`CompressionError`](ModelarDbError::CompressionError) is returned.
+    pub fn try_new(percentage: f32) -> Result<Self, ModelarDbError> {
+        if !(0.0..=100.0).contains(&percentage) || percentage.is_infinite() || percentage.is_nan() {
```

Comment:
Would the first check not also catch inf or NaN?

Reply:
Should be and the [Rust Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=cdc6623c27f475bbbcb43e6ba6c49e33) and the tests agrees.